pkg/useragent/useragent.go

```diff
@@ -61,25 +61,21 @@ var encodedUserAgent string
 // Private functions
 //
 
-// getUname tries to get a uname from the system, but not that hard. It tries
-// to execute `uname -a`, but swallows any errors in case that didn't work
-// (i.e. non-Unix non-Mac system or some other reason).
 func getUname() string {
-	path, err := exec.LookPath("uname")
-	if err != nil {
-		return unknownPlatform
-	}
+	// TODO: if there is appetite for it in the community
+	// add support for Windows GetSystemInfo
 
-	cmd := exec.Command(path, "-a") // #nosec G204
-	var out bytes.Buffer
-	cmd.Stderr = nil // goes to os.DevNull
-	cmd.Stdout = &out
-	err = cmd.Run()
-	if err != nil {
-		return unknownPlatform
+	if runtime.GOOS != "windows" {
+		u := new(unix.Utsname)
+		err := unix.Uname(u)
+		if err != nil {
+			panic(err)
+		}
+
+		return fmt.Sprintf("%s %s %s %s %s", u.Sysname, u.Nodename, u.Release, u.Version, u.Machine)
 	}
 
-	return out.String()
+	return ""
```

Comment:
Instead of returning nothing could we try to get the hostname for windows?

Reply:
`uname` is used to get information about the OS/kernel, not the hostname. Windows doesn't have a great/easy to use equivalent.